packages/lib/src/document.ts

```diff
@@ -14,7 +16,7 @@ export const formatDidDoc = (data: t.DocumentData): t.DidDocument => {
       context.push(info.context)
     }
     verificationMethods.push({
-      id: `#${keyid}`,
+      id: `${data.did}#${keyid}`,
```

Comment:
Is this actually required? I thought it was implicit if not provided?

I'm not opposed to including tho

Reply:
does this need to be done for the service `id` as well?

Reply:
I'd really rather not and just stick with the fragment for atproto purposes, but have re-read the DID core spec carefully, and gotten feedback from several other folks that this exact field (`id` on `verifiedMethod`, but not `id` in other contexts) should include the DID and the fragment together. In the spec it explicitly calls this out as needing to be a full "DID URL" which includes a `did:` prefix, the identifier, then an optional fragment.

The other `id` fields are find to be just fragments. And IIRC it is allowed to refer to the `verificationMethod` via just the fragment.